sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -117,6 +117,7 @@ private object DateTimeFormatterHelper {
       pattern: String): DateTimeFormatterBuilder = {
     val builder = createBuilder()
     pattern.split("'").zipWithIndex.foreach {
+      case ("", idx) if idx != 0 => builder.appendLiteral("'")
```

Comment:
can we add some comment to explain why `idx != 0`?

Reply:
comment add, thanks.